pandas/io/parsers.py

```diff
@@ -786,17 +786,25 @@ def read(self, nrows=None):
             return ret
 
         # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+        index, columns, mappings, col_dict = self._create_index(ret)
 
-        df = DataFrame(col_dict, columns=columns, index=index)
+        if not mappings:
```

Comment:
Until I can see / achieve otherwise, renaming like what you did below does not suffice for more complex renaming set ups, and keeping track of those changes will require "gymnastics" as well.